File: 2186-count-vowel-substrings-of-a-string/2186-count-vowel-substrings-of-a-string.rb
# @param {String} word
# @return {Integer}
def count_vowel_substrings(word)
  vowels = ['a', 'e', 'i', 'o', 'u']
  n = word.length

  (0...n).sum do |start|
    (start...n).count do |end_pos|
      substring = word[start..end_pos]
      substring.match?(/\A[aeiou]+\z/) && vowels.all? { |vowel| substring.include?(vowel) }
    end
  end
end

File: 2186-count-vowel-substrings-of-a-string/2186-count-vowel-substrings-of-a-string.py
class Solution:
    def countVowelSubstrings(self, word: str) -> int:
        vowels = {'a', 'e', 'i', 'o', 'u'}
        n=len(word)
        count=0
        for start in range(n):
            for end_pos in range(start,n):
                substring=word[start:end_pos+1]
                print(substring)
                if set(substring).issubset(vowels) and vowels.issubset(substring):
                    count+=1
                    print(count)

        return count
        